### PR TITLE
refactor(dcs): discard old DCS resources

### LINKED ISSUE #1496: Query a list of DCS flavors
I’d like to query a list of available DCS flavors.
data source: huaweicloud_dcs_flavors

### PR DESCRIPTION
**Pull request by chengxiangdong**
**What this PR does / why we need it**:

- Deprecated `huaweicloud_dcs_az` and `huaweicloud_dcs_product`.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #1496

**Special notes for your reviewer**:

- This PR needs to be approved together with https://github.com/huaweicloud/terraform-provider-huaweicloud/pull/1547.

**Release note**:

```release-note
huaweicloud_dcs_instance optimized and upgraded, discarded huaweicloud_dcs_product and huaweicloud_dcs_az.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/deprecated' TESTARGS='-run TestAccDcsInstancesV1_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/deprecated -v -run TestAccDcsInstancesV1_basic -timeout 360m -parallel 4
=== RUN   TestAccDcsInstancesV1_basic
=== PAUSE TestAccDcsInstancesV1_basic
=== CONT  TestAccDcsInstancesV1_basic
--- PASS: TestAccDcsInstancesV1_basic (150.28s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/deprecated        150.369s
```

```
make testacc TEST='./huaweicloud/services/acceptance/deprecated' TESTARGS='-run TestAccDcsProductV1DataSource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/deprecated -v -run TestAccDcsProductV1DataSource_basic -timeout 360m -parallel 4
=== RUN   TestAccDcsProductV1DataSource_basic
=== PAUSE TestAccDcsProductV1DataSource_basic
=== CONT  TestAccDcsProductV1DataSource_basic
--- PASS: TestAccDcsProductV1DataSource_basic (34.57s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/deprecated        34.642s
````

```
make testacc TEST='./huaweicloud/services/acceptance/deprecated' TESTARGS='-run TestAccDcsAZV1DataSource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/deprecated -v -run TestAccDcsAZV1DataSource_basic -timeout 360m -parallel 4
=== RUN   TestAccDcsAZV1DataSource_basic
=== PAUSE TestAccDcsAZV1DataSource_basic
=== CONT  TestAccDcsAZV1DataSource_basic
--- PASS: TestAccDcsAZV1DataSource_basic (39.04s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/deprecated        39.140s
```

### CODE REVIEW

File: docs/data-sources/dcs_az.md [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
-subcategory: "Distributed Cache Service"
+subcategory: "Deprecated"
 ---
 
 # huaweicloud_dcs_az
 
-Use this data source to get the ID of an available Huaweicloud dcs az.
+Use this data source to get the ID of an available HuaweiCloud dcs az.
+
+!> **WARNING:** It has been deprecated, use `huaweicloud_availability_zones` instead.
```
Comment (niuzhenguo) [inline docs/data-sources/dcs_az.md]: can we get ID information from `huaweicloud_availability_zones`?

Comment (chengxiangdong) [reply docs/data-sources/dcs_az.md]: no, this data source is used for the `available_zones` of the `huaweicloud_dcs_instance` resource. 
Now `available_zones` has been deprecated and this data source is no longer used.

I've change the descriptions

File: docs/data-sources/dcs_product.md [hunk 2]
```diff
@@ -6,6 +6,9 @@ subcategory: "Distributed Cache Service"
```
Comment (niuzhenguo) [inline docs/data-sources/dcs_product.md:6]: update the subcategory to `Deprecated` as well

Comment (chengxiangdong) [reply docs/data-sources/dcs_product.md:6]: done

**Pull request merged**